src/main/java/com/salesforce/einsteinbot/sdk/client/BasicChatbotClientImpl.java

```diff
@@ -86,11 +96,34 @@ void setHealthApi(HealthApi healthApi) {
     this.healthApi = healthApi;
   }
 
+  @VisibleForTesting
+  void setVersionsApi(VersionsApi versionsApi) {
+    this.versionsApi = versionsApi;
+  }
+
   @Override
   public BotResponse startChatSession(RequestConfig config,
       ExternalSessionId sessionId,
       BotSendMessageRequest botSendMessageRequest) {
 
+    String currentApiVersion;
+    try {
+      currentApiVersion = getCurrentApiVersion();
+      SupportedVersions versions = getSupportedVersions();
+      List<SupportedVersionsVersions> supportedVersions = versions.getVersions()
+          .stream()
+          .filter(v -> {
+            assert v.getVersionNumber() != null;
+            return v.getVersionNumber().equals(currentApiVersion);
+          })
+          .collect(Collectors.toList());
+      if (supportedVersions.size() != 1) {
+        throw new RuntimeException("Unable to start chat. API version not supported.");
```

Comment:
Do we want to do this check only for `startChatsession` or to all other methods like `sendMessage` and `endSession` ?

Reply:
@atulkc What are your thoughts on this? My thought process was that if the check is done when session is starting, then that is enough

Reply:
Can you add more details to the exception? It should mention the version being currently used by the sdk and the versions that are supported. Anyone looking at the stack trace and exception message should be able to know exactly what they need to do